compiler/test_intern_string.cpp

```diff
@@ -0,0 +1,16 @@
+#include "intern_string.h"
+#include <assert.h>
+
+int main(int argc, char const *argv[]) {
+  const char *input_1 = "test";
+  const char *input_2 = "test 2";
+
+  const char *address_1 = intern_string(input_1);
+  const char *address_2 = intern_string(input_2);
+
+  assert(address_1 == intern_string(input_1));
+  assert(address_2 == intern_string(input_2));
+  assert(address_1 != intern_string(input_2));
+  assert(address_2 != intern_string(input_1));
```

Comment:
The last 2 are not much useful but the other are good! There's not much to test in here because of most of the functionality is being handled by already tested code (e.g. `strlen`, `vector` etc.). But what other edge cases can you think of?